src/main/kotlin/event/player/ItemFrameInteract.kt

```diff
@@ -0,0 +1,23 @@
+package event.player
+
+import org.bukkit.Material
+import org.bukkit.entity.Entity
+import org.bukkit.entity.ItemFrame
+import org.bukkit.event.player.PlayerInteractEntityEvent
+
+object ItemFrameInteract {
+
+    fun itemframeInteractEvent(event: PlayerInteractEntityEvent) {
+        val entity: Entity? = event.rightClicked
```

Comment:
Null isn't being checked here, could even be combined with the cast, removing a whole if block below. A quick and easy solution could be:
```suggestion
        val itemFrame = event.rightClicked as? ItemFrame ?: run { return }
```

Reply:
This is a combination of [kotlins safe cast](https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator) and an early return.